GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -2694,17 +2694,24 @@ private void ViewFileHistoryMenuItem_Click(object sender, EventArgs e)
 
         private void Message_KeyUp(object sender, KeyEventArgs e)
         {
+            // Ctrl + Shift + Enter = Commit and Push
+            if (e.Control && e.Shift && e.KeyCode == Keys.Enter)
+            {
+                ExecuteCommitCommand(push: true);
+                e.Handled = true;
+            }
+
             // Ctrl + Enter = Commit
-            if (e.Control && e.KeyCode == Keys.Enter)
+            else if (e.Control && e.KeyCode == Keys.Enter)
             {
                 ExecuteCommitCommand();
```

Comment:
You could remove the first `if` block, keeping the old code and just changing this line to:

```c#
                ExecuteCommitCommand(push: e.Shift);
```

Reply:
PR updated, thanks for the recommendation!